pkg/netceptor/packetconn_test.go

```diff
@@ -0,0 +1,298 @@
+package netceptor_test
+
+import (
+	"context"
+	"errors"
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ansible/receptor/pkg/logger"
+	"github.com/ansible/receptor/pkg/netceptor"
+	"github.com/ansible/receptor/pkg/netceptor/mock_netceptor"
+	"github.com/ansible/receptor/pkg/utils"
+	"github.com/golang/mock/gomock"
+)
+
+// TestNewPacketConn tests the NewPacketConn method.
+func TestNewPacketConn(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	ctx := context.Background()
+
+	mockNetceptorForPacketConn := mock_netceptor.NewMockNetcForPacketConn(ctrl)
+	mockNetceptorForPacketConn.EXPECT().MaxForwardingHops().Return(byte(1))
+	mockNetceptorForPacketConn.EXPECT().Context().Return(context.Background())
+	mockNetceptorForPacketConn.EXPECT().GetUnreachableBroker().Return(utils.NewBroker(ctx, reflect.TypeOf(netceptor.UnreachableNotification{})))
+	mockNetceptorForPacketConn.EXPECT().GetListenerRegistry().Return(map[string]*netceptor.PacketConn{})
+
+	t.Run("NewPacketConn Success", func(t *testing.T) {
+		pc := netceptor.NewPacketConn(mockNetceptorForPacketConn, "test", 0)
+		if pc == nil {
+			t.Error("Expected new PacketConn, got nil")
+		}
+	})
+}
+
+// TestListenPacket tests the ListenPacket method.
+func TestListenPacket(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	listenPacketTestCases := []struct {
+		name             string
+		service          string
+		expectedErr      string
+		failedTestString string
+	}{
+		{"Success", "test", "", "Expected no error, but got: %v"},
+		{"Service Too Long Error", "123456789", "service name 123456789 too long", "Expected service name too long error, but got %v"},
+		{"Service Already Listening Error", "ping", "service ping is already listening", "Expected service ping is already listening, but got: %v"},
+	}
+
+	for _, testCase := range listenPacketTestCases {
+		ctx := context.Background()
+		netc := netceptor.New(ctx, "node")
+
+		t.Run(testCase.name, func(t *testing.T) {
+			_, err := netc.ListenPacket(testCase.service)
+			if testCase.expectedErr == "" && err != nil {
```

Comment:
This bank of `if` statements is repeated in the next set of tests, can they be combined into a function?